src/ukrdc_cupid/core/general.py

```diff
@@ -0,0 +1,58 @@
+from ukrdc_cupid.core.store.keygen import mint_new_pid, mint_new_ukrdcid
+
+from sqlalchemy.orm import Session
+from ukrdc_cupid.core.match.identify import identify_patient_feed, read_patient_metadata, identify_across_ukrdc
+from ukrdc_cupid.core.store.insert import insert_incoming_data
+from ukrdc_cupid.core.investigate.create_investigation import get_patients
+from ukrdc_xsdata.ukrdc import PatientRecord
+
+def process_file(xml_object:PatientRecord, session:Session):
+    """The code for the api will look very similar to this I think. 
+    until that is written it will have to sit in the utilities.
+
+    Args:
+        xml_object (PatientRecord): _description_
+        session (Session): _description_
+
+    Returns:
+        _type_: _description_
+    """
+    
+    # Attempt to identify patient on same feed
+    patient_info = read_patient_metadata(xml_object)
+    pid, ukrdcid, investigation = identify_patient_feed(session=session, patient_info=patient_info)
```

Comment:
What if the pid or ukrdcid were already the subject of an unresolved Investigation?

Reply:
It would create a new issue. That is the philosophy behind the association table in datamodel. You can just run a query to group all the issues attached to a particular ukrdcid or pid.

Reply:
I meant more if the incoming file would not itself cause an "investigation" but there had been a previous file which had caused an issue which was yet to be marked as resolved.

Reply:
Ah so you are saying we might want to add a step to the validation/patient identification to do a lookup and divert if the patient is already under investigation. I wonder if we might want to store the MRN too.

Reply:
It may depends on what the issue is but I can see a case to "quarantine" a record so we're not making it any worse by adding more data - or indeed confusing the person trying to investigate it.

Reply:
I think my assumption was if something fails it would continue to fail until its fixed but its probably sensible to make that a bit more explicit.